Lombiq.Tests.UI/ui-testing-toolkit.mjs

```diff
@@ -0,0 +1,132 @@
+    }
+}
+
+/**
+ * Navigates the browser to the given URL and then verifies that the page has loaded with non-empty content.
+ * @param {WebDriver} driver The driver whose current tab should be navigated.
+ * @param {string} url The target URL.
+ * @param {number} maxAttempts The maximum number of attempts. If exceeded, an error is thrown.
+ * @returns {Promise<void>} Success when a non-empty page has been reached.
+ */
+async function navigate(driver, url, maxAttempts = 10) {
+    for (let i = 0; i < maxAttempts; i++) {
+        await driver.navigate().to(url);
+
+        await driver.wait(() => driver
+            .executeScript('return document.readyState')
+            .then((readyState) => readyState === 'complete'));
+
+        try {
+            if ((await driver.findElement(By.xpath('//body')).getAttribute('innerHTML'))?.trim()) {
+                console.log(`Successfully reached ${url}.`);
+                return;
+            }
+        }
+        catch (exception) {
+            // Nothing to do here, let's try again.
+        }
+    }
+
+    throw new Error(`Failed to navigate to a non-empty page at ${url} in ${maxAttempts} attempts.`)
+}
+
+/**
+ * Executes a test by connecting to an existing web driver using the information in the command line arguments.
+ * @param {function(WebDriver, string):Promise<void>} test
+ * @param {function(chrome.Options):chrome.Options} configureOptions Update the configuration before the driver is built.
+ * @returns {Promise<void>} Success if the driver is created and the test has run to completion.
+ */
+async function runTest(test, configureOptions = null) {
+    const args = process.argv.slice(2);
+    if (args.length !== 3) throw new Error('Usage: node script.js driverPath startUrl tempDirectory');
+    const [driverPath, startUrl, tempDirectory] = args;
+
+    let options = new chrome.Options()
+        .addArguments('disable-dev-shm-usage')
+        .addArguments('unsafely-disable-devtools-self-xss-warnings')
+        .addArguments('disable-search-engine-choice-screen')
+        .addArguments('--lang=en-US')
+        .addArguments('disable-accelerated-2d-canvas')
+        .addArguments('disable-gpu')
+        .addArguments('force-color-profile=sRGB')
+        .addArguments('force-device-scale-factor=1')
+        .addArguments('high-dpi-support=1')
+        .addArguments('disable-smooth-scrolling')
+        .addArguments('ignore-certificate-errors')
+        .addArguments('--ignore-certificate-errors')
+        .addArguments('--no-sandbox')
+        ;
+
+    if (process.env.GITHUB_ENV) options = options.addArguments('headless');
+    if (configureOptions) options = configureOptions(options) ?? options;
+
+    const service = new chrome.ServiceBuilder(driverPath).build();
+    const driver = chrome.Driver.createSession(options, service);
+    await driver.manage().setTimeouts({ implicit: 10000 });
```

Comment:
This isn't an implicit wait, right?

Reply:
It is an [implicit wait](https://www.selenium.dev/documentation/webdriver/waits/). Why?

Reply:
Hmm, I remembered that as a wait before every interaction (i.e. that it'll click slowly). It's not that, so OK then.